devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java

```diff
@@ -245,14 +237,26 @@ public void startDev() {
             } else {
                 throw new GradleException("Unsupported DevModeMain artifact URL:" + classFile);
             }
-            addToClassPaths(classPathManifest, classPath, path);
+            addToClassPaths(classPathManifest, context, path);
 
             //now we need to build a temporary jar to actually run
 
             File tempFile = new File(getBuildDir(), extension.finalName() + "-dev.jar");
             tempFile.delete();
             tempFile.deleteOnExit();
 
+            StringBuilder resources = new StringBuilder();
+            String res = null;
+            for (File file : extension.resourcesDir()) {
+                if (resources.length() > 0)
+                    resources.append(File.pathSeparator);
+                resources.append(file.getAbsolutePath());
+                res = file.getAbsolutePath();
+            }
+            DevModeContext.ModuleInfo moduleInfo = new DevModeContext.ModuleInfo(getSourceDir().getAbsolutePath(),
```

Comment:
getSourceDir() is not actually not "correct" atm. The QuarkusPluginExtension.sourceDir/outputDirectory() methods returns a File, but it's "misleading"/wrong. The SourceSets they extract those locations from return a FileCollection. When we send it as a String/File the actual content of it is "correct" when the FileCollection only contained one item, else it's will be: file1:file2:file3....
I think we should change the Extension methods to return a List<File> instead.

Reply:
Would you be able to look into this after the initial maven support is merged? I have not really looked at Gradle at all.

Reply:
yup, we can do that!